recipes/spdlog/all/conanfile.py

```diff
@@ -82,35 +85,47 @@ def _disable_werror(self):
         tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "utils.cmake"), "/WX", "")
 
     def build(self):
+        if tools.Version(self.version) < "1.7" and tools.Version(self.deps_cpp_info["fmt"].version) >= 7:
+            raise ConanInvalidConfiguration("The project spdlog/{} requires fmt <7.x".format(self.version))
+
         self._disable_werror()
-        if self.options.header_only:
-            tools.patch(**self.conan_data["patches"][self.version])
-        cmake = self._configure_cmake()
-        cmake.build()
+        if not self.options.header_only:
+            cmake = self._configure_cmake()
+            cmake.build()
 
     def package(self):
         self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
-        cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "spdlog", "cmake"))
+        if self.options.header_only:
+            self.copy(pattern="*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        else:
+            cmake = self._configure_cmake()
+            cmake.install()
+            tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "spdlog", "cmake"))
 
     def package_id(self):
         if self.options.header_only:
             self.info.header_only()
 
     def package_info(self):
         if self.options.header_only:
-            self.cpp_info.defines = ["SPDLOG_FMT_EXTERNAL"]
+            component_name = "spdlog_header_only"
+        else:
+            component_name = "libspdlog"
+            self.cpp_info.components["libspdlog"].libs = tools.collect_libs(self)
+            self.cpp_info.components["libspdlog"].defines.append("SPDLOG_COMPILED_LIB")
+            
+        self.cpp_info.components[component_name].defines.append("SPDLOG_FMT_EXTERNAL")
+        if self.options["fmt"].header_only:
```

Comment:
Shouldn't this be other way around? I get
```
Component 'fmt::fmt-header-only' not found in 'fmt' package requirement
ERROR: Component 'fmt::fmt-header-only' not found in 'fmt' package requirement
```

Reply:
Your fmt package should be outdated. You can run conan install -U <ref>, or remove remove from cache and install again.

Reply:
This didn't go away even after bumping fmt from 6.2.1 to the latest (7.somthing).
I think it's being addressed in #2243